src/proto/beats.proto

```diff
@@ -10,22 +11,69 @@ enum Source {
   USER = 2;
 }
 
-message Beat {
+message BpmMarker {
+  optional int32 beat_index = 1;
+  optional double bpm = 2;
+  optional Source source = 3;
+}
+
+message TimeSignature {
+  optional int32 beats_per_bar = 1 [ default = 4];
+  optional int32 note_value = 2 [ default = 4 ];
+}
+
+message TimeSignatureMarker {
+  optional int32 downbeat_index = 1;
+  optional TimeSignature time_signature = 2;
+}
+/*
+message PhraseMarker {
+  optional int32 downbeat_index = 1;
+}
+
+enum SectionType {
+  OTHER = 0;
+  INTRO = 1;
+  OUTRO = 2;
+}
+
+message Color {
+  optional int32 red = 1;
+  optional int32 green = 2;
+  optional int32 blue = 3;
+}
+
+message SectionMarker {
+  optional int32 downbeat_index = 1;
+  optional string name = 2;
+  optional SectionType type = 3;
+  optional Color color = 4;
+}
+*/
+message Beats {
+  optional double first_frame_position = 1;
```

Comment:
I don't understand why this exists. Why isn't the position of the first beat sufficient?

Reply:
It is exactly what this encodes. Did you mean something else?

Reply:
I think `first_beat_frame` would be a clearer name.